sql: don't require newline at EOF during COPY

madelynnblue opened the pull request:
The comment being removed about blindly using ReadBytes may have been
incorrect. I'm not able to think through a case where it makes sense,
nor can I find a test case that fails when removing that check. And
there are some easy-to-produce failures with that check present.

Fixes #13847

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/13858)
<!-- Reviewable:end -->

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[pkg/sql/copy.go, line 158 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13858#-Ke5gzsMG_fQcAPKdP7G:-Ke5gzsMG_fQcAPKdP7H:b7akie2) ([raw file](https://github.com/cockroachdb/cockroach/blob/2f6d4177d880f9be76f27a6508f03125b93c8925/pkg/sql/copy.go#L158)):*
> ```Go
> 	for buf.Len() > 0 {
> 		line, err := buf.ReadBytes(lineDelim)
> 		if err != nil && err != io.EOF {
> ```

can you nest this instead of using `else if`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13858#-:-Ke5h2k6rw1GtB-ihGrB:bnfp4nl)*
<!-- Sent from Reviewable.io -->

tamird commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[pkg/sql/copy.go, line 158 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13858#-Ke5gzsMG_fQcAPKdP7G:-Ke5hAGcyKLbA4om5fzi:b-2o4y85) ([raw file](https://github.com/cockroachdb/cockroach/blob/2f6d4177d880f9be76f27a6508f03125b93c8925/pkg/sql/copy.go#L158)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

can you nest this instead of using `else if`?
</blockquote></details>

```
if err != nil {
  if err != io.EOF {
    return ...
  }
} else {
...
}
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13858)*
<!-- Sent from Reviewable.io -->

madelynnblue committed:
sql: don't require newline at EOF during COPY

The comment being removed about blindly using ReadBytes may have been
incorrect. I'm not able to think through a case where it makes sense,
nor can I find a test case that fails when removing that check. And
there are some easy-to-produce failures with that check present.

Fixes #13847

mjibson commented:
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/sql/copy.go, line 158 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13858#-Ke5gzsMG_fQcAPKdP7G:-Ke5it_QQmYSUxGEGQOB:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/2f6d4177d880f9be76f27a6508f03125b93c8925/pkg/sql/copy.go#L158)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

```
if err != nil {
  if err != io.EOF {
    return ...
  }
} else {
...
}
```
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13858)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13858)*
<!-- Sent from Reviewable.io -->

jordanlewis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13858#-:-Ke5jzs4Vqq4lTtZlwYt:bnfp4nl)*
<!-- Sent from Reviewable.io -->
